frontend/src/employee-frontend/components/messages/MessageEditor.tsx

```diff
@@ -159,14 +160,21 @@ export default React.memo(function MessageEditor({
     function updateSelectedReceiversOnReceiverTreeChanges() {
       const selected = getSelected(receiverTree)
       const recipientAccountIds = getSelectedBottomElements(receiverTree)
-      setMessage((old) => ({
-        ...old,
-        recipientAccountIds,
-        recipientIds: selected.map((s) => s.value),
-        recipientNames: selected.map((s) => s.label)
-      }))
+      if (
+        !isEqual(
+          new Set(message.recipientIds),
+          new Set(selected.map((option) => option.value))
+        )
+      ) {
```

Comment:
Is this actually needed to fix the problem? This adds a nasty two way dependency, because you are updating the same state that triggers the effect.

Reply:
Or is this just prevent an infinite loop because now the other effect relies on recipients too?